pkg/utils/packet.go

```diff
@@ -156,12 +156,34 @@ func SendUnsolicitedNeighborAdvertisement(srcIP net.IP, linkObj netlink.Link) er
 	// Set the destination IPv6 address to the IPv6 link-local all nodes multicast address (ff02::1).
 	var r [16]byte
 	copy(r[:], net.IPv6linklocalallnodes.To16())
-	sockAddr := syscall.SockaddrInet6{Addr: r}
-	if err := syscall.Sendto(soc, icmpv6Bytes, 0, &sockAddr); err != nil {
-		return fmt.Errorf("failed to send Unsolicited Neighbor Advertisement for IPv6 %s on Interface %s: %v", srcIP.String(), linkObj.Attrs().Name, err)
-	}
+	destSockAddr := syscall.SockaddrInet6{Addr: r}
 
-	return nil
+	var nextSleepDuration = 10 * time.Millisecond
+
+	for {
+		err2 := syscall.Sendto(soc, icmpv6Bytes, 0, &destSockAddr)
+
+		if err2 == nil {
+			return nil
+		}
+
+		errno, ok := err2.(syscall.Errno)
+		if ok && errno == syscall.EADDRNOTAVAIL {
+			// Assume the error happened because the IPv6 source address is
+			// still tentative. Retry?
+			if retryUntil.Before(time.Now()) {
```

Comment:
This condition looks flipped to me. 

```suggestion
			if time.Now().Before(retryUntil) {
```

Any chance to get a unit test on this function?

Reply:
This seems hard to unit test.